### PR TITLE
users v1.0

### PR DESCRIPTION
**Pull request by Jeremy-Marzuolo**

### CODE REVIEW

File: src/app/user/user.component.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+<form [formGroup]="form" (ngSubmit)="submit()">
+
+  <mat-form-field>
+    <mat-label for="name">Name</mat-label>
+    <input matInput formControlName="name" type="text" id="name">
+  </mat-form-field>
+
+  <mat-form-field>
+    <mat-label for="age">Age</mat-label>
+    <input matInput formArrayName="age" type="number" id="age" min="1">
```
Comment (GrandSchtroumpf) [inline src/app/user/user.component.html]: Maybe you can add a max too. Just to avoid 100000

File: src/app/user/user.component.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      <mat-option value="france"> France </mat-option>
+      <mat-option value="spain"> Spain </mat-option>
+      <mat-option value="italy"> Italy </mat-option>
+      <mat-option value="greece">Greece</mat-option>
+    </mat-select>
+  </mat-form-field>
+
+  <mat-form-field>
+    <mat-label>Gender</mat-label>
+    <mat-select formControlName="gender">
```
Comment (GrandSchtroumpf) [inline src/app/user/user.component.html]: Prefer radio button or button-toggle when working with few options like here.

File: src/app/user/user.component.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  form = new FormGroup({
+    name: new FormControl(),
+    age: new FormControl(),
+    nationality: new FormControl(),
+    gender: new FormControl()
+  })
+
+  infos?: User[] = []
+
+  constructor() { }
```
Comment (GrandSchtroumpf) [inline src/app/user/user.component.ts]: remove constructor & ngOnIni if you don't use it.

File: src/app/user/user.component.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  ngOnInit(): void {
+  }
+
+  submit(){
+    if (this.form.valid){
+      this.infos?.push(this.form.value)
+      console.log(this.infos)
+    }
+    this.form.reset()
```
Comment (GrandSchtroumpf) [inline src/app/user/user.component.ts]: Hmm careful. If the form is invalid, you'll reset it...

**Review by GrandSchtroumpf — CHANGES_REQUESTED** [review]

File: src/app/user/user.component.html (hunk 1 above)
Comment (Jeremy-Marzuolo) [reply src/app/user/user.component.html]: Done

File: src/app/user/user.component.ts (hunk 4 above)
Comment (Jeremy-Marzuolo) [reply src/app/user/user.component.ts]: Done

File: src/app/user/user.component.ts (hunk 3 above)
Comment (Jeremy-Marzuolo) [reply src/app/user/user.component.ts]: Done

File: src/app/user/user.component.html (hunk 2 above)
Comment (Jeremy-Marzuolo) [reply src/app/user/user.component.html]: Done

File: src/app/genders.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export const genders = {
+  male: "Male",
+  female: "Female",
+  transgenre: "Transgenre",
+  non_binary: "Non Binary",
```
Comment (GrandSchtroumpf) [inline src/app/genders.ts]: use camelCase

File: src/app/user/user.component.html [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    <input matInput formControlName="name" type="text" id="name">
+  </mat-form-field>
+
+  <mat-form-field>
+    <mat-label for="age">Age</mat-label>
+    <input matInput formArrayName="age" type="number" id="age" min="1" max="120">
+  </mat-form-field>
+
+
+  <!-- <mat-form-field>
```
Comment (GrandSchtroumpf) [inline src/app/user/user.component.html]: remove comments

File: src/app/user/user.component.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    nationality: new FormControl(),
+    gender: new FormControl()
+  })
+
+  infos?: User[] = []
+
+  submit(){
+    if (this.form.valid){
+      this.infos?.push(this.form.value)
+      console.log(this.infos)
```
Comment (GrandSchtroumpf) [inline src/app/user/user.component.ts]: remove console.log

**Review by GrandSchtroumpf — APPROVED** [review]

File: src/app/genders.ts (hunk 5 above)
Comment (Jeremy-Marzuolo) [reply src/app/genders.ts]: Done

File: src/app/user/user.component.html (hunk 6 above)
Comment (Jeremy-Marzuolo) [reply src/app/user/user.component.html]: Done

File: src/app/user/user.component.ts (hunk 7 above)
Comment (Jeremy-Marzuolo) [reply src/app/user/user.component.ts]: Done and oh my god you have a good eye

Comment (Jeremy-Marzuolo) [reply src/app/user/user.component.ts]: is this your job ?!

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**